### PR TITLE
add RR details to ECR Metadata

### PR DESCRIPTION
**Pull request by BobanL**
# PULL REQUEST

## Summary
- EcrViewer
  - Refactor EcrMetadata page to follow the same pattern to accept DisplayData[] rather than evaluating in the file itself.
  - Update seed data to include RR Routing Entity organization and link that to `rr-reportability-information-observation` as a performer - [FHIR docs](https://build.fhir.org/ig/HL7/case-reporting/StructureDefinition-rr-reportability-information-observation.html#:~:text=performer%3AroutingEntity,for%20the%20observation)
  - Add an Bundle with ECR Metadata for unit tests. Also add unit tests for Patient address to ensure that is empty if nothing is available. 
  - Refactor `formatAddress` to return empty string if no address is provided. Previously it would Return `\n, \n`
  - Fix codesmell to return value rather than setting variable first 
- Fhir-Converter
  - Set emails if the string contains an '@' symobl or 'mailto:'.  Some of the data we had for organizations didn't have 'mailto:' in the emails, but all emails will include '@' symbol and phone numbers will not. 
  - In `Header.Liquid` include RR Routing Entity Organization to be the entity that the data will be routed to. Routing Entities will have the [`RR7` code](https://phinvads.cdc.gov/vads/ViewCodeSystemConcept.action?oid=2.16.840.1.114222.4.5.232&code=RR7). 
  - Update `ObservationRRCondition` to be able to handle multiple reportability rules. Originally, I thought it could only have one, but I have found [examples of multiple rules for a single observation](https://github.com/CDCgov/phdi/pull/1092#:~:text=Example%20of%20multiple%20conditions).
    - Also add performer to link to RoutingEntity 
  - Add Organization RR Routing Entity [based on the http://hl7.org/fhir/us/ecr/StructureDefinition/rr-routing-entity-organization spec](https://build.fhir.org/ig/HL7/case-reporting/StructureDefinition-rr-routing-entity-organization.html) 
  

- A new PR will be opened to convert this data into a table

## Related Issue
part of #1012

## Additional Information
- now:
image
- before:
image

- Example of multiple conditions
```XML
<observation classCode="OBS" moodCode="EVN">
	<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.3.12"/>
	<id root="f7106df9-08a6-4708-8511-388e59ea9a4d"/>
	<code code="64572001" codeSystem="2.16.840.1.113883.6.96" codeSystemName="SNOMED" displayName="Condition">
		<translation code="75323-6" codeSystem="2.16.840.1.113883.6.1" codeSystemName="LOINC" displayName="Condition"/>
	</code>
	<value code="840539006" codeSystem="2.16.840.1.113883.6.96" codeSystemName="SNOMED CT" displayName="Disease caused by severe acute respiratory syndrome coronavirus 2 (disorder)" xsi:type="CD"/>
	<entryRelationship typeCode="COMP">
		<organizer classCode="CLUSTER" moodCode="EVN">
			<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.3.13"/>
			<id root="648663f5-2373-4260-884c-f83e7a7fa673"/>
			<code code="RRVS5" codeSystem="2.16.840.1.114222.4.5.274" codeSystemName="Location Relevance" displayName="Patient home address"/>
			<statusCode code="completed"/>
			<participant typeCode="LOC">
				<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.4.3"/>
				<participantRole>
					<id extension="lac" root="2.16.840.1.113883.4.6"/>
					<code code="RR12" codeSystem="2.16.840.1.114222.4.5.232" codeSystemName="PHIN Questions" displayName="Rules Authoring Agency"/>
					<addr use="WP">
						<streetAddressLine/>
						<city/>
						<state/>
						<postalCode nullFlavor="NI"/>
					</addr>
					<telecom nullFlavor="NI"/>
					<playingEntity>
						<name>Los Angeles County Department of Public Health</name>
					</playingEntity>
				</participantRole>
			</participant>
			<participant typeCode="LOC">
				<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.4.1"/>
				<participantRole>
					<id extension="lac" root="2.16.840.1.113883.4.6"/>
					<code code="RR7" codeSystem="2.16.840.1.114222.4.5.232" codeSystemName="PHIN Questions" displayName="Routing Entity"/>
					<addr use="WP">
						<streetAddressLine/>
						<city/>
						<state/>
						<postalCode nullFlavor="NI"/>
					</addr>
					<playingEntity>
						<name>Los Angeles County Department of Public Health</name>
					</playingEntity>
				</participantRole>
			</participant>
			<component typeCode="COMP">
				<observation classCode="OBS" moodCode="EVN">
					<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.3.19"/>
					<id root="0cd644b4-eb0b-49de-bec8-f16fd85cf15b"/>
					<code code="RR1" codeSystem="2.16.840.1.114222.4.5.232" codeSystemName="PHIN Questions" displayName="Determination of reportability"/>
					<value code="RRVS1" codeSystem="2.16.840.1.114222.4.5.274" codeSystemName="PHIN VS (CDC Local Coding System)" displayName="Reportable" xsi:type="CD"/>
					<entryRelationship typeCode="RSON">
						<observation classCode="OBS" moodCode="EVN">
							<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.3.27"/>
							<id root="cfdd2088-2310-446b-8e9c-4354de519e80"/>
							<code code="RR3" codeSystem="2.16.840.1.114222.4.5.232" codeSystemName="PHIN Questions" displayName="Determination of reportability rule"/>
							<value xsi:type="ST">COVID-19 (as a diagnosis or active problem)</value>
						</observation>
					</entryRelationship>
					<entryRelationship typeCode="RSON">
						<observation classCode="OBS" moodCode="EVN">
							<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.3.27"/>
							<id root="ad681ff9-7ee5-4778-96e0-82e14a464a03"/>
							<code code="RR3" codeSystem="2.16.840.1.114222.4.5.232" codeSystemName="PHIN Questions" displayName="Determination of reportability rule"/>
							<value xsi:type="ST">Detection of SARS-CoV-2 nucleic acid in a clinical or post-mortem specimen by any method</value>
						</observation>
					</entryRelationship>
				</observation>
			</component>
			<component typeCode="COMP">
				<act classCode="ACT" moodCode="EVN">
					<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.3.20"/>
					<id root="3cde8472-bdff-4635-aa3c-e47c4e4b5983"/>
					<code code="RRVS13" codeSystem="2.16.840.1.114222.4.5.274" codeSystemName="PHIN VS (CDC Local Coding System)" displayName="Outbreak- or Cluster related"/>
					<priorityCode code="RRVS15" codeSystem="2.16.840.1.114222.4.5.274" codeSystemName="PHIN VS (CDC Local Coding System)" displayName="Information only"/>
					<reference typeCode="REFR">
						<externalDocument classCode="DOC" moodCode="EVN">
							<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.3.17"/>
							<code nullFlavor="OTH">
								<originalText>CDC COVID-19 webpage</originalText>
							</code>
							<text mediaType="text/html">
								<reference value="https://www.cdc.gov/coronavirus/2019-ncov/index.html"/>
							</text>
						</externalDocument>
					</reference>
				</act>
			</component>
			<component typeCode="COMP">
				<act classCode="ACT" moodCode="EVN">
					<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.3.20"/>
					<id root="3791407b-0455-40fa-9662-2ac243257ac0"/>
					<code code="RRVS12" codeSystem="2.16.840.1.114222.4.5.274" codeSystemName="PHIN VS (CDC Local Coding System)" displayName="PHA Contact Information"/>
					<priorityCode code="RRVS15" codeSystem="2.16.840.1.114222.4.5.274" codeSystemName="PHIN VS (CDC Local Coding System)" displayName="Information only"/>
					<reference typeCode="REFR">
						<externalDocument classCode="DOC" moodCode="EVN">
							<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.3.17"/>
							<code nullFlavor="OTH">
								<originalText>Los Angeles County Department of Public Health Office of the Chief Medical Informatics Officer IRIS System 241 North Figueroa Street Los Angeles, CA 90012 213-288-7696</originalText>
							</code>
							<text mediaType="text/html">
								<reference value="http://publichealth.lacounty.gov/"/>
							</text>
						</externalDocument>
					</reference>
				</act>
			</component>
			<component typeCode="COMP">
				<observation classCode="OBS" moodCode="EVN">
					<templateId extension="2017-04-01" root="2.16.840.1.113883.10.20.15.2.3.14"/>
					<id root="ca0528f9-25fc-47b8-a18a-a32282f1cd79"/>
					<code code="RR4" codeSystem="2.16.840.1.114222.4.5.232" codeSystemName="PHIN Questions" displayName="Timeframe to report (urgency)"/>
					<value unit="min" value="0" xsi:type="PQ">
						<translation code="88694003" codeSystem="2.16.840.1.113883.6.96" codeSystemName="SNOMED CT" displayName="Immediate (qualifier value)"/>
					</value>
				</observation>
			</component>
		</organizer>
	</entryRelationship>
</observation>
```
## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

File: containers/ecr-viewer/src/app/utils.tsx [hunk 1]
```diff
@@ -39,10 +40,10 @@ export const extractPatientAddress = (
     fhirBundle,
     fhirPathMappings.patientStreetAddress,
   );
-  const city = evaluate(fhirBundle, fhirPathMappings.patientCity);
-  const state = evaluate(fhirBundle, fhirPathMappings.patientState);
-  const zipCode = evaluate(fhirBundle, fhirPathMappings.patientZipCode);
-  const country = evaluate(fhirBundle, fhirPathMappings.patientCountry);
+  const city = evaluate(fhirBundle, fhirPathMappings.patientCity)[0];
```
Comment (nickbristow) [inline containers/ecr-viewer/src/app/utils.tsx:43]: these are arrays? Is there a chance of having multiples of cities for example? Do we need to account for that?

File: containers/ecr-viewer/src/app/utils.tsx [hunk 2]
```diff
@@ -68,25 +69,35 @@ export const extractFacilityAddress = (
...
-  }`;
+  let address = {
+    streetAddress: streetAddress || [],
+    cityState: [city, state],
+    zipCodeCountry: [zipCode, country],
+  };
+
+  return [
+    address.streetAddress.join("\n"),
+    address.cityState.filter(Boolean).join(", "),
```
Comment (nickbristow) [inline containers/ecr-viewer/src/app/utils.tsx:96]: Is this filter for just type `Boolean`?

**Review by nickbristow — APPROVED** [review]
Just some questions about some stuff

File: containers/fhir-converter/Templates/eCR/DataType/_ContactPoint.liquid [hunk 3]
```diff
@@ -3,7 +3,7 @@
 		"system":"phone",
 		"value": "{{ContactPoint.value | replace: 'tel:\s*', ''}}",
 		"use": "{{ ContactPoint.use | get_property: 'ValueSet/TelecomUse' }}",
-	{% elsif ContactPoint.value contains 'mailto:' -%}
+	{% elsif ContactPoint.value contains 'mailto:' or ContactPoint.value contains '@' -%}
```
Comment (emmastephenson) [inline containers/fhir-converter/Templates/eCR/DataType/_ContactPoint.liquid:6]: Nice catch! 🐟

File: containers/fhir-converter/Templates/eCR/Header.liquid [hunk 4]
```diff
@@ -46,7 +46,20 @@
             {% else -%}
               {% assign observationId = ob.value | to_json_string | generate_uuid -%}
             {% endif -%}
-            {% include 'Resource/ObservationRRCondition' observationEntry: ob, ID: observationId -%}
+
+            {% if ob.entryRelationship and ob.entryRelationship.organizer and ob.entryRelationship.organizer %}
```
Comment (emmastephenson) [inline containers/fhir-converter/Templates/eCR/Header.liquid:50]: I think you're checking the same value twice here - should this be checking null or participants or something?

```suggestion
            {% if ob.entryRelationship and ob.entryRelationship.organizer and ob.entryRelationship.organizer.participant %}
```

File: containers/ecr-viewer/src/app/view-data/components/EcrSummary.tsx [hunk 5]
```diff
@@ -123,11 +123,7 @@ const EcrSummary = ({ fhirPathMappings, fhirBundle }: EcrViewerProps) => {
                 <h4>Reportable Condition</h4>
               </div>
               <div className="grid-col-fill">
-                {
-                  evaluate(fhirBundle, fhirPathMappings.rrDisplayName)[0].split(
-                    "-",
-                  )[1]
-                }
+                {evaluate(fhirBundle, fhirPathMappings.rrDisplayNames)[0]}
```
Comment (emmastephenson) [inline containers/ecr-viewer/src/app/view-data/components/EcrSummary.tsx:126]: Not at all in scope for this PR, but just a note for the future that I think when we integrate with NBS, they'll probably pass along both an eICR ID and the condition a user is viewing the document for. When that happens, instead of grabbing the first condition we'll instead need to find whichever condition the user is trying to view, and highlight that one in the eCR Summary. 

Again wildly out of scope for today - just a note!

**Review by emmastephenson — COMMENTED** [review]
This looks great, thank you!

File: containers/ecr-viewer/src/app/utils.tsx (hunk 2 above)
Comment (BobanL) [reply containers/ecr-viewer/src/app/utils.tsx:96]: This filter is to filter `falsey` values. This is to prevent if like city or state were null it would try to join [empty string with empty string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join#:~:text=The%20string%20conversions%20of%20all%20array%20elements%20are%20joined%20into%20one%20string.%20If%20an%20element%20is%20undefined%2C%20null%2C%20it%20is%20converted%20to%20an%20empty%20string%20instead%20of%20the%20string%20%22null%22%20or%20%22undefined%22.).

File: containers/ecr-viewer/src/app/utils.tsx (hunk 1 above)
Comment (BobanL) [reply containers/ecr-viewer/src/app/utils.tsx:43]: Evaluate returns an array `any[]`, which you can see in the demo - https://hl7.github.io/fhirpath.js/.
Now I changed that format address function to specify the types (strings) this just needed to be any[]

File: containers/ecr-viewer/src/app/view-data/components/EcrSummary.tsx (hunk 5 above)
Comment (BobanL) [reply containers/ecr-viewer/src/app/view-data/components/EcrSummary.tsx:126]: Neat! So would they give us a name you think or something else to identify the condition?

**Comment by BobanL** [comment]
@sarahtress, when you have a chance can you do a design review given the screenshots in the main post? Thanks!

File: containers/ecr-viewer/src/app/view-data/components/EcrSummary.tsx (hunk 5 above)
Comment (emmastephenson) [reply containers/ecr-viewer/src/app/view-data/components/EcrSummary.tsx:126]: Yep! Probably either the name or SNOMED code for the condition.

**Comment by BobanL** [comment]
Merging this as is and then another PR will be out to convert this data into table.
